connectors/src/lib/models/microsoft.ts

```diff
@@ -110,12 +122,12 @@ MicrosoftConfigurationRoot.init(
     ],
   }
 );
-ConnectorModel.hasMany(MicrosoftConfigurationRoot);
+ConnectorModel.hasMany(MicrosoftConfigurationRootModel);
 
 // MicrosftResource stores files/folders/channels and other resources synced from Microsoft.
-export class MicrosoftResource extends Model<
-  InferAttributes<MicrosoftResource>,
-  InferCreationAttributes<MicrosoftResource>
+export class MicrosoftResourceModel extends Model<
```

Comment:
Maybe we could name it differently than "resource"? to avoid it clashing with the ConnectorResource pattern